src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsClientModelValidatorProvider.cs

```diff
@@ -46,14 +68,14 @@ public void GetValidators(ClientValidatorProviderContext context)
                 DataAnnotationsClientModelValidationFactory factory;
                 if (_attributeFactories.TryGetValue(attribute.GetType(), out factory))
                 {
-                    context.Validators.Add(factory(attribute));
+                    context.Validators.Add(factory(attribute, stringLocalizer));
```

Comment:
Its in `DataAnnotationsClientModelValidator<TAttribute>` class `GetErrorMessage` method. Call `stringLocalizer` if it is not null else just format the error message